ext/reflection/php_reflection.c

```diff
@@ -2488,6 +2488,37 @@ ZEND_METHOD(reflection_parameter, getClass)
 }
 /* }}} */
 
+/* {{{ proto public string ReflectionParameter::getClassName()
+   Returns this parameters's class name or null if this there is none */
+ZEND_METHOD(reflection_parameter, getClassName)
+{
+	reflection_object *intern;
+	parameter_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	if (param->arg_info->class_name) {
+		const char *class_name;
+		size_t class_name_len;
+
+		if (param->fptr->type == ZEND_INTERNAL_FUNCTION) {
+			class_name = ((zend_internal_arg_info*)param->arg_info)->class_name;
+			class_name_len = strlen(class_name);
+
+			RETURN_STRINGL(class_name, class_name_len);
+		} else {
+			class_name = param->arg_info->class_name->val;
+			class_name_len = param->arg_info->class_name->len;
+
+			RETURN_STR(param->arg_info->class_name);
```

Comment:
Missing a copy here. Also the assignments above it are redundant.

    RETURN_STR(zend_string_copy(param->arg_info->class_name));
    // OR
    ZVAL_STR_COPY(return_value, param->arg_info->class_name);